File: Orimath.ViewPlugins/Controls/LoadExtension.cs
using System;
using System.Windows;
using System.Windows.Markup;

namespace Orimath.Controls
{
    public class LoadExtension : MarkupExtension
    {
        public string Uri { get; set; }

        public LoadExtension() => Uri = "";

        public LoadExtension(string uri) => Uri = uri;

        public override object ProvideValue(IServiceProvider serviceProvider)
        {
            return Application.LoadComponent(new Uri(Uri, UriKind.Relative));
        }
    }
}

File: Orimath.ViewPlugins/Controls/LoadExtension.fs
namespace Orimath.Controls
open System
open System.Windows
open System.Windows.Markup

type LoadExtension(uri: string) =
    inherit MarkupExtension()
    member val Uri = uri with get, set

    override this.ProvideValue(_) =
        Application.LoadComponent(Uri(this.Uri, UriKind.Relative))

    new() = LoadExtension("")
